hal: nxp: Check Kconfig before including HAL

yonsch opened the pull request:
hal_nxp was added to cmake's include directories globally, without checking if the hal is enabled. This made all Zephyr builds end up including hal_nxp, even for other vendors.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>

manuargue commented on modules/hal_nxp/CMakeLists.txt:
```diff
@@ -4,7 +4,9 @@
 
-add_subdirectory(${ZEPHYR_CURRENT_MODULE_DIR} hal_nxp)
-add_subdirectory_ifdef(CONFIG_USB_DEVICE_DRIVER usb)
+if(CONFIG_USE_HAL_NXP)
```
can we use `if(CONFIG_HAS_MCUX OR CONFIG_HAS_IMX_HAL OR CONFIG_HAS_S32_HAL)` instead of defining a new Kconfig option? This is already being done here https://github.com/zephyrproject-rtos/hal_nxp/blob/master/CMakeLists.txt#L1

yonsch committed:
hal: nxp: Check Kconfig before including HAL

hal_nxp was added to cmake's include directories globally, without
checking if the hal is enabled. This made all Zephyr builds end up
including hal_nxp, even for other vendors.

Signed-off-by: Yonatan Schachter <yonatan.schachter@gmail.com>
